scrapyjs/middleware.py

```diff
@@ -43,24 +52,26 @@ def from_crawler(cls, crawler):
         return cls(crawler, splash_base_url, slot_policy)
 
     def process_request(self, request, spider):
-        splash_options = request.meta.get('splash')
+        splash_options = self.get_splash_options(request, spider)
         if not splash_options:
             return
 
+        elif request.meta.get("_splash_processed"):
+            return
+
         if request.method != 'GET':
             log.msg("Currently only GET requests are supported by SplashMiddleware; %s "
                     "will be handled without Splash" % request, logging.WARNING)
             return request
 
         meta = request.meta
-        del meta['splash']
-        meta['_splash_processed'] = splash_options
 
         slot_policy = splash_options.get('slot_policy', self.slot_policy)
         self._set_download_slot(request, meta, slot_policy)
 
         args = splash_options.setdefault('args', {})
-        args.setdefault('url', request.url)
+        args['url'] = request.url
```

Comment:
Maybe I was thinking of allowing user not to use 'url' argument at all. This argument is not required for Splash scripts - e.g. you can render a chunk of HTML using splash:set_content. But the meta syntax doesn't make much sense in this case; something in lines of https://github.com/scrapinghub/scrapyjs/issues/12 could be a better fit.